src/components/AppBreadcrumbs/AppBreadcrumbs.tsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import { withStyles } from '@material-ui/core/styles';
+import { Breadcrumbs, Link } from '@material-ui/core';
+import { Link as RouterLink } from 'react-router-dom';
+
+function capitalize(str: string): string {
+  return str.charAt(0).toUpperCase() + str.slice(1);
+}
+
+const styles = (theme: any) => ({
+  breadcrumb: {
+    fontStyle: 'italic',
+  },
+  terminalBreadcrumb: {
+    fontStyle: 'italic',
+    color: theme.palette.primary.dark,
+    cursor: 'default',
+    '&:hover': {
+      textDecoration: 'none',
+    },
+  },
+});
+
+export type Context = { type: string; id: string; name: string };
+export type BreadcrumbElement = { text: string; to: string };
+
+type AppBreadcrumbsProps = {
+  links: Array<BreadcrumbElement>;
+  context: Context;
+};
+
+const BreadcrumbsLink = withStyles(styles)((props: any) => {
+  const { classes, ...other } = props;
+  return <Link component={RouterLink} color="primary" className={classes.breadcrumb} {...other} />;
+});
+
+const TerminalBreadcrumb = withStyles(styles)((props: any) => {
+  const { classes, ...other } = props;
+  return (
+    <Link
+      component={RouterLink}
+      color="primary"
+      className={classes.terminalBreadcrumb}
+      {...other}
+      onClick={(e) => e.preventDefault()}
```

Comment:
I don't think you need this do you?

Reply:
You do. If you don't include it, the React Router will still register the location change and mess with history.